lerobot/scripts/server/actor_server.py

```diff
@@ -47,131 +44,101 @@
-    """
+from lerobot.scripts.server.utils import get_last_item_from_queue
 
-    def __init__(self, transition=None, interaction_message=None):
-        self.transition = transition
-        self.interaction_message = interaction_message
+ACTOR_SHUTDOWN_TIMEOUT = 30
 
 
 def receive_policy(
-    learner_client: hilserl_pb2_grpc.LearnerServiceStub,
+    cfg: DictConfig,
+    parameters_queue: Queue,
     shutdown_event: Event,
-    parameters_queue: queue.Queue,
 ):
     logging.info("[ACTOR] Start receiving parameters from the Learner")
-    bytes_buffer = io.BytesIO()
-    step = 0
-    try:
-        for model_update in learner_client.StreamParameters(hilserl_pb2.Empty()):
-            if shutdown_event.is_set():
-                logging.info("[ACTOR] Shutting down policy streaming receiver")
-                return hilserl_pb2.Empty()
-
-            if model_update.transfer_state == hilserl_pb2.TransferState.TRANSFER_BEGIN:
-                bytes_buffer.seek(0)
-                bytes_buffer.truncate(0)
-                bytes_buffer.write(model_update.parameter_bytes)
-                logging.info("Received model update at step 0")
-                step = 0
-                continue
-            elif (
-                model_update.transfer_state == hilserl_pb2.TransferState.TRANSFER_MIDDLE
-            ):
-                bytes_buffer.write(model_update.parameter_bytes)
-                step += 1
-                logging.info(f"Received model update at step {step}")
-            elif model_update.transfer_state == hilserl_pb2.TransferState.TRANSFER_END:
-                bytes_buffer.write(model_update.parameter_bytes)
-                logging.info(
-                    f"Received model update at step end size {bytes_buffer_size(bytes_buffer)}"
-                )
-
-                state_dict = torch.load(bytes_buffer)
-
-                bytes_buffer.seek(0)
-                bytes_buffer.truncate(0)
-                step = 0
 
-                logging.info("Model updated")
+    # Setup process handlers to handle shutdown signal
+    # But use shutdown event from the main process
+    setup_process_handlers()
 
-                parameters_queue.put(state_dict)
+    learner_client, grpc_channel = learner_service_client(
+        host=cfg.actor_learner_config.learner_host,
+        port=cfg.actor_learner_config.learner_port,
+    )
 
+    try:
+        iterator = learner_client.StreamParameters(hilserl_pb2.Empty())
+        receive_bytes_in_chunks(
+            iterator,
+            parameters_queue,
+            shutdown_event,
+            log_prefix="[ACTOR] parameters",
+        )
     except grpc.RpcError as e:
         logging.error(f"[ACTOR] gRPC error: {e}")
 
-    return hilserl_pb2.Empty()
+    grpc_channel.close()
+    logging.info("[ACTOR] Received policy loop stopped")
 
 
-def transitions_stream(shutdown_event: Event, message_queue: queue.Queue):
+def transitions_stream(
+    shutdown_event: Event, transitions_queue: Queue
```

Comment:
Got this info from pylance for `Event` `Variable not allowed in type expression Pylance` maybe we should mute pylance for this line